walnuts_cpp/include/walnuts/nuts.hpp

```diff
@@ -198,26 +200,35 @@ Vec<S> transition(Random<S, Generator> &rng, const F &logp_grad_fun,
   Span<S> span_accum(std::move(theta), std::move(rho), std::move(grad), logp);
   for (Integer depth = 0; depth < max_depth; ++depth) {
     bool go_forward = rng.uniform_binary();
-    bool uturn_flag;
     if (go_forward) {
-      Span<S> span_next = build_span<Direction::Forward>(
-          rng, logp_grad_fun, inv_mass, step, depth, span_accum, uturn_flag);
-      if (uturn_flag)
+      constexpr Direction D = Direction::Forward;
+
+      auto maybe_next_span =
+          build_span<D>(rng, logp_grad_fun, inv_mass, step, depth, span_accum);
+      if (!maybe_next_span)
         break;
```

Comment:
braces on ifs

Reply:
Is there a clang-format rule we could use to enforce this uniformly?

Reply:
I'm on the other side of most people on this one.  It was half and half, so I went and removed them all.  I'm not adamant about anything formatting related, so if you think it's worth somehow configuring clang-format to do it, go for it (but please write doc so I can figure out how to use it).